### PR TITLE
Add documents on running distributedlog in distributed mode

### PR DESCRIPTION
**Pull request by sijie**
- add documentation on running distributedlog in distributed mode
- improve scripts to start daemons for zookeeper, bookkeeper and write proxy
- misc changes on stats provider and write proxy announcement.

### CODE REVIEW

File: distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java [hunk 1]
```diff
@@ -33,19 +36,43 @@
...
+                try {
+                    port = Integer.parseInt(hostAndPort[1]);
+                } catch (NumberFormatException nfe) {
+                    logger.warn("Failed to retrieve zookeeper server port from {}. Use default port 2181.",
+                            zkServer, nfe);
+                }
+            }
+            InetSocketAddress address = InetSocketAddress.createUnresolved(host, port);
+            zkAddresses[i] = address;
+            ++i;
```
Comment (mgodave) [inline distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java]: you can get rid of the i index and replace the array with an ArrayList appending as you go. ImmutableList.copyOf can take a Collection

File: distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java [hunk 2]
```diff
@@ -33,19 +36,43 @@
...
         return uri.getAuthority().replace(";", ",");
     }
 
+    private static Iterable<InetSocketAddress> getZkAddresses(URI uri) {
+        String zkServers = getZKServersFromDLUri(uri);
+        String[] zkServerList = StringUtils.split(zkServers, ',');
+        InetSocketAddress[] zkAddresses = new InetSocketAddress[zkServerList.length];
+        int i = 0;
+        for (String zkServer : zkServerList) {
+            String[] hostAndPort = StringUtils.split(zkServer, ':');
```
Comment (mgodave) [inline distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java]: Guava already exists and has a HostAndPort.fromString("...").withDefaultPort(2181)

File: distributedlog-service/src/main/java/com/twitter/distributedlog/service/DistributedLogServerApp.java [hunk 3]
```diff
@@ -88,16 +94,37 @@ private void run() {
...
+            } catch (ConfigurationException e) {
+                throw new IllegalArgumentException("Failed to load distributedlog configuration from " + configFile + ".");
+            } catch (MalformedURLException e) {
+                throw new IllegalArgumentException("Failed to load distributedlog configuration from malformed "
+                        + configFile + ".");
+            }
+        }
+        Optional<String> statsProviderCls = getOptionalStringArg(cmdline, "pd");
+        // load the stats provider
+        final StatsProvider statsProvider;
```
Comment (mgodave) [inline distributedlog-service/src/main/java/com/twitter/distributedlog/service/DistributedLogServerApp.java]:
```
final StatsProvider statsProvider = getOptionalStringArg(cmdline, "pd")
  .transform(new Function<String, StatsProvider>() {
    @override 
    public StatsProvider apply(String name) {
      return ReflectionUtils.newInstance(name.get(), StatsProvider.class);
    }  
  }).or(new NullStatsProvider());
```

**Comment by mgodave** [comment]
I think a lot of the codahale metrics libs moved to the dropwizard namespace. I don't think the other stuff is supported any longer.

https://github.com/dropwizard/metrics

The old stuff is not a Go library that points you towards dropwizard

https://github.com/codahale/metrics

File: distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java (hunk 1 above)
Comment (mgodave) [inline distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java]: Even better, build an ImmutableList using the Builder and eliminate the copy.

Comment (sijie) [inline distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java]: yup. that's a good suggestion.

File: distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java (hunk 2 above)
Comment (sijie) [inline distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java]: that's a very helpful util

File: distributedlog-service/src/main/java/com/twitter/distributedlog/service/DistributedLogServerApp.java (hunk 3 above)
Comment (sijie) [inline distributedlog-service/src/main/java/com/twitter/distributedlog/service/DistributedLogServerApp.java]: cool

**Comment by sijie** [comment]
yup. the bookkeeper provider uses an old version. we could consider bumping the version later.

**Comment by sijie** [comment]
@mgodave addressed your comments

**Pull request merged**